lib/moon/components/as_compoent.ex

```diff
@@ -37,10 +36,9 @@ defmodule Moon.Components.AsComponent do
         {=@role}
         {...@attrs}
         {=@id}
-        data-testid={@testid}
       ><#slot /></button>
     {#elseif @as == "a"}
-      <a {=@class} {=@href} {...@attrs} {=@id} data-testid={@testid}><#slot /></a>
+      <a {=@class} {=@href} {...@attrs} {=@id}><#slot /></a>
```

Comment:
do not like the idea to remove obvious way. does moving data-testid before @attrs in list of atributes solve your problem?

Reply:
I browsed the repository and places that are adding `testid`.

It seems that all other components are using it as:

<img width="241" alt="Screenshot 2023-03-31 at 11 39 59" src="https://user-images.githubusercontent.com/3199333/229085205-bd9c6b52-684e-469e-872d-37f4d4ffee2f.png">

Nothing uses `testid` directly. I can change that of course. Let me know.

Reply:
ok, here is suitable solution:
`<a {=@class} {=@href} {=@id} {...Map.merge(%{"data-testid": @testid}, @attrs)}><#slot /></a>`
and add than the same for the button

Reply:
But this will cause a bug, when a user provides `testid` in both places, won't it?

Reply:
why bug? attrs.data-testid will overwrite testid.

Reply:
Exactly.

What about:


```elixir

      attrs={
        "data-testid": "xyz"
      },
      testid: "abc"
```
?

Reply:
will be used "xyz"
what is an expected (non-buggy) behaviour for you in this case?

Reply:
The ideal scenario?
To keep the component as is and change other components that are using it.
I can change that.
Wdyt?

Reply:
that makes sence. if you wish - let it be